maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java

```diff
@@ -280,24 +281,12 @@ public DefaultGraphBuilder( BuildResumptionDataRepository buildResumptionDataRep
         if ( !requiredSelectors.isEmpty() || !optionalSelectors.isEmpty() )
         {
             Set<MavenProject> excludedProjects = new HashSet<>( requiredSelectors.size() + optionalSelectors.size() );
-            excludedProjects.addAll( getProjectsBySelectors( request, projects, requiredSelectors, true ) );
-            excludedProjects.addAll( getProjectsBySelectors( request, projects, optionalSelectors, false ) );
+            List<MavenProject> allProjects = graph.getAllProjects();
+            excludedProjects.addAll( getProjectsBySelectors( request, allProjects, requiredSelectors, true ) );
+            excludedProjects.addAll( getProjectsBySelectors( request, allProjects, optionalSelectors, false ) );
 
             result = new ArrayList<>( projects );
-
-            for ( MavenProject excludedProject : excludedProjects )
-            {
-                boolean isExcludedProjectRemoved = result.remove( excludedProject );
-
-                if ( isExcludedProjectRemoved )
-                {
-                    List<MavenProject> children = excludedProject.getCollectedProjects();
-                    if ( children != null )
-                    {
-                        result.removeAll( children );
-                    }
-                }
-            }
```

Comment:
If you are wondering why this has been removed, this is because we made a mistake during an earlier merge. The children are already collected in the `getProjectsBySelectors` method which is called on line 285 and 286. The removed code would be redundant.